util/fs/fs_test.go

```diff
@@ -51,24 +58,29 @@ func TestFSDetectorResize(t *testing.T) {
 	if runtime.GOOS != "linux" {
 		t.Skip("TestFSDetectorResize skipped because OS is not 'linux'")
 	}
-	if !hasPasswordlessSudo() {
-		t.Skip("TestFSDetectorResize skipped because password-less sudo is required and not presently available")
+	if !hasRoot() {
+		t.Skip("TestFSDetectorResize skipped because sudo privileges are not currently available")
 	}
 
 	// Non-existent test case.
 	if fsType, err := Detect("/TestFSDetectorResize/foo/bar/baz"); err != ErrNoFilesystemDetected {
-		t.Errorf("Expected ErrNoFilesystemDetected error from Detect, but got result fsType=%s err=%+v", fsType, err)
+		t.Errorf("Expected ErrNoFilesystemDetected error from Detect, but got result fsType=%v err=%+v", fsType, err)
 	}
 
 	// Non-existent test case.
-	if err := Resize("/TestFSDetectorResize/foo/bar/baz"); err == nil  {
+	if err := Resize("/TestFSDetectorResize/foo/bar/baz"); err == nil {
 		t.Error("Expected error from Resize, but got nothing")
 	}
 
-	for _, targetFsType := range supportedFsTypes {
+	for _, targetFSType := range supportedFsTypes {
+		if !strings.HasPrefix(targetFSType, "ext") {
+			t.Logf("Skipping resize test for fsType=%[1]s because resize2fs only supports ext2/ext3/ext4", targetFSType)
+			continue
+		}
```

Comment:
change this to check the same way `TestDeviceFormatter` is testing today for `supportedFsTypes` list